src/resource.ts

```diff
@@ -620,9 +630,18 @@ export class BaseResource implements IResource {
           }
         }
       }
+      // strip unneeded metadata
+      console.log('input', input[inputName])
+      let data = input[inputName]
+      if (data?.__typename) {
+        /// test for object
+        const { __typename, ...rest } = input[inputName]
```

Comment:
as a simple workaround i'd remove __typename here if input[inputName] is an object (test can be nicer of course)

would that be okay for you @jukart  or can you think of a nicer solution that is easy to implement